src/commands/dev/index.js

```diff
@@ -35,47 +34,59 @@ const startStaticServer = async ({ settings, log }) => {
   log(`\n${NETLIFYDEVLOG} Server listening to`, settings.frameworkPort)
 }
 
+const NOT_FOUND_EXIT_CODE = 127
+
+const isNonExistingCommandError = (error) => {
+  // `ENOENT` is only returned for non Windows systems
+  // See https://github.com/sindresorhus/execa/pull/447
+  // exitCode === 127 is returned from npm/yarn
+  if (error.code === 'ENOENT' || error.exitCode === NOT_FOUND_EXIT_CODE) {
+    return true
+  }
+
+  // this only works on English versions of Windows
+  return (
+    typeof error.message === 'string' && error.message.includes('is not recognized as an internal or external command')
+  )
+}
+
 const startFrameworkServer = async function ({ settings, log, exit }) {
   if (settings.noCmd) {
     return await startStaticServer({ settings, log })
   }
 
   log(`${NETLIFYDEVLOG} Starting Netlify Dev with ${settings.framework || 'custom config'}`)
-  const commandBin = await which(settings.command).catch((error) => {
-    if (error.code === 'ENOENT') {
-      throw new Error(
-        `"${settings.command}" could not be found in your PATH. Please make sure that "${settings.command}" is installed and available in your PATH`,
+
+  const commandWithArgs = `${settings.command} ${settings.args.join(' ')}`
+  // we use reject=false to avoid rejecting synchronously when the command doesn't exist
+  // we can't try->await->catch since we don't want to block on the framework server which
+  // is a long running process
+  const frameworkProcess = execa.command(commandWithArgs, { preferLocal: true, reject: false })
+  frameworkProcess.stdout.pipe(stripAnsiCc.stream()).pipe(process.stdout)
+  frameworkProcess.stderr.pipe(stripAnsiCc.stream()).pipe(process.stderr)
+  process.stdin.pipe(frameworkProcess.stdin)
+
+  // eslint-disable-next-line promise/catch-or-return,promise/prefer-await-to-then
+  frameworkProcess.then(async () => {
+    const result = await frameworkProcess
+    const { exitCode = 0 } = result
+    // eslint-disable-next-line promise/always-return
+    if (result instanceof Error && isNonExistingCommandError(result)) {
+      log(
+        NETLIFYDEVERR,
+        `Failed launching framework server. Please verify ${chalk.magenta(`'${settings.command}' exists`)}`,
+      )
+    } else {
+      log(
+        exitCode > 0 ? NETLIFYDEVERR : NETLIFYDEVWARN,
+        `"${commandWithArgs}" exited with code ${exitCode}. Shutting down Netlify Dev server`,
```

Comment:
The `error` rejected by `execa` contains some useful information beyond just the exit code, available in [`error.message`](https://github.com/sindresorhus/execa#all) or [`error.shortMessage`](https://github.com/sindresorhus/execa#shortmessage). Would we want to show it?

Reply:
Done in https://github.com/netlify/cli/pull/2857/commits/0fd3527fad9bfd82cecba703188a832b82672e69.

`shortMessage` has enough information so we don't need to format the error ourselves. The other case is that the long running process exited successfully, so we need to format an error manually.